packages/backend/server/src/plugins/indexer/job.ts

```diff
@@ -107,4 +125,49 @@ export class IndexerJob {
     await this.queue.remove(workspaceId, 'indexer.indexWorkspace');
     await this.service.deleteWorkspace(workspaceId);
   }
+
+  @OnJob('indexer.autoIndexWorkspaces')
+  async autoIndexWorkspaces(payload: Jobs['indexer.autoIndexWorkspaces']) {
+    const startSid = payload.lastIndexedWorkspaceSid ?? 0;
+    const workspaces = await this.models.workspace.findManyBySid(startSid, 100);
+    if (workspaces.length === 0) {
+      return JOB_SIGNAL.Repeat;
+    }
+    let addedCount = 0;
+    for (const workspace of workspaces) {
+      if (workspace.indexed) {
+        continue;
+      }
+      await this.queue.add(
+        'indexer.indexWorkspace',
+        { workspaceId: workspace.id },
+        { jobId: workspace.id }
+      );
+      addedCount++;
+    }
+    const nextSid = workspaces[workspaces.length - 1].sid;
+    this.logger.log(
+      `Auto added ${addedCount} workspaces to queue, lastIndexedWorkspaceSid: ${startSid} -> ${nextSid}`
+    );
+
+    // update the lastIndexedWorkspaceSid in the payload and repeat the job after 30 seconds
+    payload.lastIndexedWorkspaceSid = nextSid;
```

Comment:
@forehalo  Reconstitute a pure job to pass the data, and add the `indexed` property to determine whether to update the index fully